src/vdaf/hits.rs

```diff
@@ -0,0 +1,897 @@
+    x: F,
+}
+
+enum SketchState {
+    Ready,
+    RoundOne,
+    RoundTwo,
+}
+
+impl<F: FieldElement> PrepareStep<Vec<F>> for HitsPrepareStep<F> {
+    type Input = Vec<F>;
+    type Output = Vec<F>;
+
+    fn step<M: IntoIterator<Item = Vec<F>>>(
+        mut self,
+        inputs: M,
+    ) -> PrepareTransition<Self, Vec<F>> {
+        match self.state {
+            SketchState::Ready => {
+                let inputs: Vec<_> = inputs.into_iter().collect();
+                if !inputs.is_empty() {
+                    return PrepareTransition::Fail(VdafError::Uncategorized(format!(
+                        "expected no messages, got {}",
+                        inputs.len()
+                    )));
+                }
+
+                let z_share = self.z.to_vec();
+
+                self.state = SketchState::RoundOne;
+                PrepareTransition::Continue(self, z_share)
+            }
+
+            SketchState::RoundOne => {
+                let inputs: Vec<_> = inputs.into_iter().collect();
+                if inputs.len() != 2 {
+                    return PrepareTransition::Fail(VdafError::Uncategorized(format!(
+                        "unexpected message count: got {}; want 2",
+                        inputs.len()
+                    )));
+                }
+
+                // Compute polynomial coefficients.
+                let mut z = [F::zero(); 3];
+                for z_share in inputs.iter() {
+                    if z_share.len() != 3 {
+                        // XXX Tim- How do I print the state here?
+                        return PrepareTransition::Fail(VdafError::Uncategorized(format!(
+                            "unexpected message length: got {}; want {}",
+                            z_share.len(),
+                            3
+                        )));
+                    }
+
+                    z[0] += z_share[0];
+                    z[1] += z_share[1];
+                    z[2] += z_share[2];
+                }
+
+                // Compute our share of the polynomial evaluation.
+                let y_share =
+                    vec![(self.d * z[0]) + self.e + self.x * ((z[0] * z[0]) - z[1] - z[2])];
+
+                self.state = SketchState::RoundTwo;
+                PrepareTransition::Continue(self, y_share)
+            }
+
+            SketchState::RoundTwo => {
+                let inputs: Vec<_> = inputs.into_iter().collect();
+                if inputs.len() != 2 {
+                    return PrepareTransition::Fail(VdafError::Uncategorized(format!(
+                        "unexpected message count: got {}; want 2",
+                        inputs.len()
+                    )));
+                }
+
+                // Compute the polynomial evaluation.
+                let mut y = F::zero();
+                for y_share in inputs.iter() {
+                    if y_share.len() != 1 {
+                        // XXX Tim- How do I print the state here?
```

Comment:
Do you mean `self.state`? Is it not in scope here?

Reply:
What I'm asking is how to print the variant of `self.state` (an `enum`) in the error message. You once pointed out how to do this, but I forgot :/

Reply:
I think you could just derive a `Debug` implementation on `SketchState` (none of its variants have fields so the implementation will just write the variant name) and then use `{:?}` in a format string.

Reply:
Oh, derp. Thanks.